lib/logic/box_communicator.dart

```diff
@@ -106,23 +104,67 @@ class BoxCommunicator {
             print('response had no id');
             continue;
           }
-          //all files of the current box have been downloaded
-          if (await box.get(id) != null) {
+
+          if (await box.get(id) != null || idList.contains(id)) {
+            //all files of the current box have been downloaded or checked (sizeLimit reached)
+            print(await box.get(id) != null);
+            print(idList.contains(id));
             print("got all files");
             break;
           }
-          await box.put(id, response.body);
-          //add size of this response to size of all data combined (for limiting data usage)
-          storage.put(
-              'totalSizeInBytes', totalSizeInBytes + response.contentLength);
-          print("new entry at: ");
-          print(DateTime.now());
+          //add the current id to the idList, so we can later check whether we had this file already
+          idList.add(id);
+
+          //for priority: split data string so that we can use the
+          // timestamp to compare values.
+          int incomingTime = jsonDecode(response.body)["timestamp"] ??
```

Comment:
it's nice that you used ?? but maybe there should be a comment to explain what it does since it might be (very likely) that future developers haven't seen it before